Sources/Qualification/QualifyingCoordinator.swift

```diff
@@ -0,0 +1,157 @@
+import Coordination
+import GDSCommon
+import Logging
+import Networking
+import SecureStore
+import UIKit
+
+/// A type that is responsible for coordinating the user's eligibility to access the main functionality of the app.
+///
+/// Performs orchestration based on the state of the app to send the user to either:
+/// - Login Coordinator: requiring the user to authenticate to enter the app.
+/// - Main Coordinator: allowing the user to access the main app functionality (i.e. Wallet)
+///
+@MainActor
+final class QualifyingCoordinator: NSObject,
+                                   ParentCoordinator,
+                                   AppQualifyingServiceDelegate {
+    private let window: UIWindow
+    var childCoordinators = [ChildCoordinator]()
+
+    private let analyticsCenter: AnalyticsCentral
+    private let appQualifyingService: QualifyingService
+    private let sessionManager: SessionManager
+    private let networkClient: NetworkClient
+
+    private var loginCoordinator: LoginCoordinator? {
+        childCoordinators.firstInstanceOf(LoginCoordinator.self)
+    }
+
+    private var mainCoordinator: MainCoordinator? {
+        childCoordinators.firstInstanceOf(MainCoordinator.self)
+    }
+
+    private lazy var unlockViewController: UnlockScreenViewController = {
+        let viewModel = UnlockScreenViewModel(analyticsService: analyticsCenter.analyticsService) { [unowned self] in
+            Task {
+                await appQualifyingService.evaluateUser()
+            }
+        }
+        return UnlockScreenViewController(viewModel: viewModel)
+    }()
+
+    init(window: UIWindow,
+         analyticsCenter: AnalyticsCentral,
+         appQualifyingService: QualifyingService,
+         sessionManager: SessionManager,
+         networkClient: NetworkClient) {
+        self.window = window
+        self.appQualifyingService = appQualifyingService
+        self.analyticsCenter = analyticsCenter
+        self.sessionManager = sessionManager
+        self.networkClient = networkClient
+        super.init()
+        self.appQualifyingService.delegate = self
+    }
+    
+    func start() {
+        lock()
+    }
+
+    func lock() {
+        displayViewController(unlockViewController)
+    }
+
+    func didChangeAppInfoState(state appInfoState: AppInformationState) {
+        switch appInfoState {
+        case .appUnconfirmed:
+            lock()
+        case .appConfirmed:
```

Comment:
what do we mean by app confirmed/ unconfirmed? As in what state is the app in here?

Reply:
App unconfirmed means we don't know what the state of the app is therefore the qualifying checks need to run and the lock screen should be displayed, this is the initial state of the `appInfoState` property in the `AppQualifyingService` before any checks have run.
App confirmed is in the inverse where we're happy that the app qualifying checks have completed successfully and we want to then move onto qualifying the user.
Does that make sense? Happy to have a chat about it or rename some stuff if that isn't immediately obvious 🙂

Reply:
Yes, makes sense - thank you!

Reply:
Should we rename these to be clearer?